lib/puppet-lint/plugins/check_conditionals.rb

```diff
@@ -5,18 +5,16 @@
 PuppetLint.new_check(:selector_inside_resource) do
   def check
     resource_indexes.each do |resource|
-      resource[:tokens].each do |token|
-        if token.type == :FARROW
-          if token.next_code_token.type == :VARIABLE
-            unless token.next_code_token.next_code_token.nil?
-              if token.next_code_token.next_code_token.type == :QMARK
-                notify :warning, {
-                  :message => 'selector inside resource block',
-                  :line    => token.line,
-                  :column  => token.column,
-                }
-              end
-            end
+      resource[:tokens].select { |token|
+        token.type == :FARROW && token.next_code_token.type == :VARIABLE
+      }.each do |token|
```

Comment:
Rather than iterate through the array twice here, I would suggest keeping the original each iterator and using `next unless token.type == :FARROW && token.next_code_token.type == :VARIABLE` at the start of the block.